### PR TITLE
feat: add `foundRows` option for MySQLi config

### PR DESCRIPTION
**Pull request by ducng99**
**Description**
Added a new database config `foundRows` for MySQLi. Enabling this adds the flag `MYSQLI_CLIENT_FOUND_ROWS` when connecting to a MySQL database.

The flag changes the behaviour of `affectedRows()` for `UPDATE` or `INSERT ... ON DUPLICATE KEY UPDATE`, where the number of rows updated to the same value will also be returned. By default, it only returns the number rows with a value changed.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by ducng99** [comment]
Update: fixed the CS issue in the test file

**Comment by ddevsr** [comment]
Looks good, by the way i see unit test in Database Live many skipped after connection to database. @kenjis I think better skipped in setUp? i can make PR later.

File: system/Database/BaseConnection.php [hunk 1]
```diff
@@ -35,6 +35,7 @@
  * @property-read string     $DSN
  * @property-read array|bool $encrypt
  * @property-read array      $failover
+ * @property-read bool       $foundRows
```
Comment (kenjis) [inline system/Database/BaseConnection.php:38]: Remove this. BaseConnection does not have it.

File: system/Database/MySQLi/Connection.php [hunk 2]
```diff
@@ -81,6 +81,16 @@ class Connection extends BaseConnection
      */
     public $numberNative = false;
 
+    /**
+     * Use MYSQLI_CLIENT_FOUND_ROWS
+     *
+     * Whether affected_rows should return number of rows found,
```
Comment (kenjis) [inline system/Database/MySQLi/Connection.php:87]:
```suggestion
     * Whether affectedRows() should return number of rows found,
```

File: user_guide_src/source/changelogs/v4.6.0.rst [hunk 3]
```diff
@@ -121,6 +121,8 @@ Forge
 Others
 ------
 
+- Added a new configuration `foundRows` for MySQLi to use `MYSQLI_CLIENT_FOUND_ROWS`.
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.6.0.rst:124]:
```suggestion
- Added a new configuration ``foundRows`` for MySQLi to use ``MYSQLI_CLIENT_FOUND_ROWS``.
```

**Comment by kenjis** [comment]
> I think better skipped in setUp?

Yes, it should be in the setUp() method.

**Comment by ducng99** [comment]
I have moved the DBDriver checks to setUp() but it would still make the connection to DB each time.
I can of course create a property like $testsSkipped and check it in each test, let me know if that's what you prefer.

Other suggested changes are applied

File: tests/system/Database/Live/MySQLi/FoundRowsTest.php [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+/**
+ * @internal
+ */
+#[Group('DatabaseLive')]
+final class FoundRowsTest extends CIUnitTestCase
+{
+    use DatabaseTestTrait;
+
+    /**
+     * @var array<string,mixed>
```
Comment (kenjis) [inline tests/system/Database/Live/MySQLi/FoundRowsTest.php:31]:
```suggestion
     * Database config for tests
     *
     * @var array<string, mixed>
```

File: tests/system/Database/Live/MySQLi/FoundRowsTest.php [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    {
+        parent::setUp();
+
+        $config = config('Database');
+
+        $this->tests = $config->tests;
+
+        if ($this->tests['DBDriver'] !== 'MySQLi') {
+            $this->markTestSkipped('Only MySQLi can complete this test.');
+        }
```
Comment (kenjis) [inline tests/system/Database/Live/MySQLi/FoundRowsTest.php:48]: Move `parent::setUp();` to the bottom.

```suggestion
        $config = config('Database');

        $this->tests = $config->tests;

        if ($this->tests['DBDriver'] !== 'MySQLi') {
            $this->markTestSkipped('Only MySQLi can complete this test.');
        }
        
        parent::setUp();
```

**Review by michalsn — APPROVED** [review]
Looks good!

**Review by kenjis — APPROVED** [review]
Thank you!

**Comment by ducng99** [comment]
Thanks everyone!

**Comment by alirahman99** [comment]
caranya biar bsa debuging instalasi pada codelnigter gimana

**Pull request merged**